강원대 FE_윤채연_1주차 과제

ychy61 opened the pull request:
안녕하세요 멘토님! 저는 강원대학교 윤채연입니다. 
피드백을 위해 시간을 내주셔서 감사합니다!

### 질문

1. **Alias 설정**
   이번 과제에서 craco를 이용하여 alias를 설정했는데, 그 외에도 alias를 설정할 수 있는 다른 방법이 있는지 궁금합니다!

2. **tsconfig의 자주 사용되는 항목**
   tsconfig에 어떤 항목들이 포함되어 있는지 확인해보았는데, 멘토님이 자주 사용되는 항목들이 어떤 것들이 있는지 알고 싶습니다.

3. **eslint와 prettier 설정 확인**
   eslint와 prettier 사용에 아직 익숙하지 않습니다. 제가 설정한 eslint와 prettier의 값에서 문제가 될 요소가 있는지 확인 부탁드립니다!

아직 많이 부족한 상태라 더 열심히 노력하겠습니다!

ychy61 committed:
feat(project): Create React App으로 프로젝트 생성

--template typescript 옵션을 통해 Typescript로 프로젝트가 동작되게
세팅함

ychy61 committed:
docs(readme): README.md 업데이트

구현한 기능에 대해서 체크해주었습니다.

ychy61 committed:
refactor(project): 불필요한 코드 및 파일 정리

프로젝트에서 사용되지 않는 코드와 파일을 정리했습니다.

ychy61 committed:
feat(webpack): 절대 경로 alias 설정
craco를 이용하여 alias 설정

ychy61 committed:
chore(tsconfig): tsconfig 설정
tsconfig 각 항목에 대한 내용을 확인하고 현재 불필요한 내용을
삭제하였으나, 추가적으로 불필요한 사항은 프로젝트를 진행하면서 파악할
수 있을 것 같습니다.

ychy61 committed:
chore(lint): Lint 룰 세팅
eslint-config-airbnb와 추가적인 설정을 해주었습니다

ychy61 committed:
feat(css): apply reset CSS

ychy61 committed:
feat(emotion): Emotion 스타일 라이브러리 설치

ychy61 committed:
chore(gitignore): settings.json 파일을 .gitignore에 추가

ychy61 committed:
docs(readme): README.md 업데이트

수행한 내용을 반영하였습니다.

ychy61 committed:
feat: Button 컴포넌트를 구현

Theme props에 따라 버튼의 컬러와 디자인이 다르게 보여지도록함
Size props에 따라 버튼의 사이즈가 다르게 보여지도록

ychy61 committed:
feat: Button 컴포넌트 수정

...rest를 사용하여 나머지 모든 속성을 Button 컴포넌트에 전달해 Button Element의 기본 속성들을 모두 사용할 수 있도록 함

ychy61 committed:
feat: Input 컴포넌트를 구현

ychy61 committed:
feat: Image 컴포넌트를 구현

ychy61 committed:
feat: Image 컴포넌트 Storybook 추가

ychy61 committed:
feat: GoodsItem 컴포넌트 구현

ychy61 committed:
feat: Grid, Container 컴포넌트를 구현

ychy61 committed:
doc: update README.md

현재 구현 내용을 반영하여 README를 수정하였습니다

ychy61 committed:
feat: update README.md

step3 질문에 대한 답 작성

ychy61 committed:
feat: Update README.md

수행한 내용에 대해서 체크하였습니다.

yujo11 commented on src/components/Button/Button.tsx:
```diff
@@ -0,0 +1,73 @@
+        return '0.75rem'; // 작은 버튼의 글꼴 크기
+      case 'large':
+        return '1.25rem'; // 큰 버튼의 글꼴 크기
+      case 'responsive':
+        return '1rem'; // 반응형 버튼의 글꼴 크기
+      default:
+        return '1rem'; // 기본 글꼴 크기
```
소소하게 아래처럼도 작성가능하니 참고하시면 좋을거 같아요

```suggestion
      case 'responsive':
      default:
        return '1rem'; // 기본 글꼴 크기
```

yujo11 commented on src/components/Button/Button.tsx:
```diff
@@ -0,0 +1,73 @@
+  cursor: pointer;
+  font-size: ${(props) => {
+    switch (props.size) {
+      case 'small':
+        return '0.75rem'; // 작은 버튼의 글꼴 크기
```
주석을 적절하게 활용하는 것은 꼭 필요하고 아주 좋은 방법입니다!

하지만 지금은 코드로도 충분히 설명이 되는 부분에도 모두 주석이 남겨져 있어 오히려 코드를 읽는 것을 방해하는 느낌도 조금 드는데요. 필요한 곳들에만 적절하게 주석을 남겨보시면 좋을거 같아요!

yujo11 commented on src/components/Button/Button.tsx:
```diff
@@ -0,0 +1,73 @@
+}
+
+// styled.button을 사용하여 버튼 스타일 정의
+const Button = styled.button<ButtonProps>`
```
`prop`의 `size`로 결정되는 style 요소가 많은데요. 이럴 때는 아래 코드에서 사용한 것과 같은 styleMapper를 만들어서 활용해보셔도 좋을거 같아요
```ts
export const _Box = styled.div(
  (props: BoxProps) => css`
    display: flex;
    gap: ${props.gap}px;
    ${alignMapper(props.direction)}
  `
);

const alignMapper = (direction: StyledBoxType['direction']) => {
  switch (direction) {
    case 'vertical':
      return css({
        flexDirection: 'column',
      });

    case 'horizontal':
    default:
      return css({
        flexDirection: 'row',
      });
  }
};
```

yujo11 commented on src/components/Image/Image.tsx:
```diff
@@ -0,0 +1,52 @@
+  position: relative;
+  overflow: hidden;
+  width: 100%;
+  padding-bottom: ${(props) => {
+    if (props.ratio === 'square') return '100%';
+    if (typeof props.ratio === 'number') return `${(1 / props.ratio) * 100}%`;
+    return '56.25%'; // 기본 16:9 비율
+  }};
```
style에 따른 분기를 줄 때 `if`문을 사용한 경우와 `switch-case` 문을 사용한 경우가 있는데 어떤걸 기준으로 사용한걸까요?

yujo11 reviewed (changes requested):
안녕하세요 채연님~~ 첫 번쨰 리뷰네요 ㅎㅎ 만나서 반갑습니다!

코드에 대한 리뷰와 더불어 한번쯤 생각해보면 좋을 내용들에 대해서도 코멘트 남겼으니 확인해보시면 좋을거 같아요!

>Alias 설정

저는 보통 설정 및 관리의 용이성을 위해 `tsconfig.json` 파일에서만 alias를 설정해서 관리하는데요. eslint로도 가능하고 방법은 다양한데 `CRA` 환경을 사용해 본 지가 너무 오래되어서 CRA에서는 어떤 방법을 사용해야 하는지까지는 조언을 드리기 어렵네요 😅

>tsconfig의 자주 사용되는 항목

옵션을 하나하나 꼼꼼하게 확인해보신게 인상적이었습니다! 기본적으로는 strict하게 타입 검사가 가능하도록 설정해두고, 위에서 언급한 path alias 같은 기능들을 많이 사용하는거 같아요!

>eslint와 prettier 설정 확인

크게 문제가 될만한 부분은 보이지 않고 잘 설정해주신거 같습니다! 웹 접근성 관련 된 plugin도 추가해보셔도 좋겠다는 생각이 들긴 하네요 ㅎㅎ

---

첫번째 미션 진행하느라 고생 많으셨습니다 채연님~~~ 👍 👍 👍

ychy61 committed:
refactor: Update README.md

프로젝트 실행 방법을 추가하였습니다

ychy61 committed:
refactor: index.html meta 태그 수정

불필요한 meta태그 삭제 및 내용을 적절하게 수정하였습니다.

ychy61 committed:
refactor: Button.tsx 수정

React.ButtonHTMLAttributes<HTMLButtonElement>를 사용하여 버튼의 모든
속성을 포함하도록 할 수 있도록 수정
styleMapper를 만들어 적용함

ychy61 commented:
> 안녕하세요 채연님~~ 첫 번쨰 리뷰네요 ㅎㅎ 만나서 반갑습니다!
> 
> 코드에 대한 리뷰와 더불어 한번쯤 생각해보면 좋을 내용들에 대해서도 코멘트 남겼으니 확인해보시면 좋을거 같아요!
> 
> > Alias 설정
> 
> 저는 보통 설정 및 관리의 용이성을 위해 `tsconfig.json` 파일에서만 alias를 설정해서 관리하는데요. eslint로도 가능하고 방법은 다양한데 `CRA` 환경을 사용해 본 지가 너무 오래되어서 CRA에서는 어떤 방법을 사용해야 하는지까지는 조언을 드리기 어렵네요 😅
> 
> > tsconfig의 자주 사용되는 항목
> 
> 옵션을 하나하나 꼼꼼하게 확인해보신게 인상적이었습니다! 기본적으로는 strict하게 타입 검사가 가능하도록 설정해두고, 위에서 언급한 path alias 같은 기능들을 많이 사용하는거 같아요!
> 
> > eslint와 prettier 설정 확인
> 
> 크게 문제가 될만한 부분은 보이지 않고 잘 설정해주신거 같습니다! 웹 접근성 관련 된 plugin도 추가해보셔도 좋겠다는 생각이 들긴 하네요 ㅎㅎ
> 
> 첫번째 미션 진행하느라 고생 많으셨습니다 채연님~~~ 👍 👍 👍

ychy61 commented:
안녕하세요 멘토님! 저는 강원대학교 윤채연입니다.
피드백을 위해 시간을 내주셔서 감사합니다!

남겨주신 피드백 정말 감사합니다!
이전에 남겨주신 피드백을 반영하여 코드를 수정했습니다.
혹시나 더 수정해야 되거나 부족한 점, 생각해봐야 할 점 있으면 말씀해주세요!!

ychy61 reviewed (commented):
멘토님 시간을 내어 피드백 해주셔서 감사합니다!
남겨주신 피드백을 반영하여 코드를 수정했습니다!
제가 깃헙 qr이 익숙하지 않아 수정 후에 코멘트를 남기게 되었는데 이렇게 하는 것이 맞는지 잘 모르겠네요..ㅎㅎ

yujo11 commented:
안녕하세요 채연님 코멘트 꼼꼼하게 확인하고 댓글 남겨주셔서 감사합니다 ㅎㅎ 고생 많으셨습니다 ! 👍 👍 👍
